File: ch07/test2.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int test(int n) {
    int a = 1;
    for (int i = n; i > 0; i--) {
        a = a + i;
    }
    return a;
}

int main(int argc, char *argv[]) {
    int n = atoi(argv[1]); // Example input
    int result = test(n);
    printf("test(%d) = %d\n", n, result);
    return 0;
}

File: ch08/test2.lua
local test = function(n,k)
    local a = 0
    for i = n, 1, -k do
        a = a + i
    end
    return a
end

print( test(arg[1],arg[2]))
-- Example usage
-- local n = 10000000 -- Example input
-- local result = fib(n)
-- print(string.format("fib(%d) = %d", n, result))
-- print(string.dump(fib))